flytekit/clients/auth/token_client.py

```diff
@@ -119,7 +119,11 @@ def get_token(
         raise AuthenticationError("Status Code ({}) received from IDP: {}".format(response.status_code, response.text))
 
     j = response.json()
-    return j["access_token"], j["expires_in"]
+    refresh_token = None
```

Comment:
One thing to note is that you can actually receive the refresh token in the `body`. In our internal implementation we add a check like this:
```python
    if refresh_token:
        body["refresh_token"] = refresh_token
```
and then later:
```python
    j = response.json()
    # Check if refresh_token already existed and was sent with the body, or if we are getting it for the first time
    if refresh_token or "refresh_token" in j and j["refresh_token"] is not None:
        return j["access_token"], j["refresh_token"], j["expires_in"]
    else:
        logging.debug("Refresh token was not received in the response, only storing access token")
        return j["access_token"], None, j["expires_in"]
```

Reply:
Not sure what you mean, wouldn't that be adding the refresh token into the request payload?

Reply:
So what this would mean is that you fetch the pre-existing refresh token if there is one, alongside the request payload.

Reply:
I see. I think I'm confused because the code that I modified that uses this currently just passes grant type `DEVICE_CODE`. The code path that I'm using to refresh the token (which I copied from the PKCE auth) just seems to make a request directly to the token API and that has worked for us.

Reply:
I think I could reuse the token client if I made that change you suggested, let me try that.

Reply:
That seems to work. Added in fe8277b45fdba47c4224b97dad5460cf36af757d